cognite/client/utils/_identifier.py

```diff
@@ -226,3 +239,31 @@ class SingletonIdentifierSequence(IdentifierSequenceCore[Identifier]):
 
 class DataModelingIdentifierSequence(IdentifierSequenceCore[DataModelingIdentifier]):
     ...
+
+
+class UserIdentifierSequence(IdentifierSequenceCore[UserIdentifier]):
+    @classmethod
+    def load(cls, user_identifiers: str | Sequence[str]) -> UserIdentifierSequence:
+        if isinstance(user_identifiers, str):
+            return cls(identifiers=[UserIdentifier(user_identifiers)], is_singleton=True)
+
+        elif isinstance(user_identifiers, Sequence):
+            return cls(identifiers=list(map(UserIdentifier, map(str, user_identifiers))), is_singleton=False)
+
+        raise TypeError(f"user_identifiers must be of type str or Sequence[str]. Found {type(user_identifiers)}")
+
+    def chunked(self, chunk_size: int) -> list[UserIdentifierSequence]:  # type: ignore [override]
+        return [
+            UserIdentifierSequence(chunk, is_singleton=self.is_singleton())
+            for chunk in split_into_chunks(self._identifiers, chunk_size)
+        ]
+
+    def assert_singleton(self) -> None:
+        if not self.is_singleton():
+            raise ValueError("Exactly one user identifier (string) must be specified")
+
+    def as_dicts(self) -> list[dict[str, str]]:  # type: ignore [override]
+        return [identifier.as_dict() for identifier in self._identifiers]
+
+    def as_primitives(self) -> list[str]:  # type: ignore [override]
+        return [identifier.as_primitive() for identifier in self._identifiers]
```

Comment:
If we introduce a new type in the core like this:
```python
T_Identifier_Type = TypeVar("T_Identifier_Type")

class IdentifierSequenceCore(Generic[T_Identifier, T_Identifier_Type], ABC):
```
and change the implementation in the parent class to 
```python
    def as_dicts(self) -> list[dict[str, T_Identifier_Type]]:
        return [identifier.as_dict() for identifier in self._identifiers]

    def as_primitives(self) -> list[T_Identifier_Type]:
        return [identifier.as_primitive() for identifier in self._identifiers]
```
Then, these methods do not need to be reimplemented. 
```suggestion
```

Reply:
Will try this one out (tomorrow) 🚀

Reply:
Raises an error from typing extensions: 🤔 
```
cognite/client/data_classes/_base.py:30: in <module>
    from cognite.client.utils._identifier import IdentifierSequence
cognite/client/utils/_identifier.py:185: in <module>
    class IdentifierSequence(IdentifierSequenceCore[Identifier]):
../../.pyenv/versions/3.8.16/lib/python3.8/typing.py:261: in inner
    return func(*args, **kwds)
../../.pyenv/versions/3.8.16/lib/python3.8/typing.py:897: in __class_getitem__
    _check_generic(cls, params)
.venv/lib/python3.8/site-packages/typing_extensions.py:164: in _check_generic
    raise TypeError(f"Too {'many' if alen > elen else 'few'} parameters for {cls};"
E   TypeError: Too few parameters for <class 'cognite.client.utils._identifier.IdentifierSequenceCore'>; actual 1, expected 2
```

Reply:
When I think about it, this doesnt make that much sense; we want it to a Generic of IdentifierCore (our protocol with as_dict/as_primitive), and adding this unbounded, would mean we could make it a generic of "whatever".

Reply:
I still removed these method overrides - at the cost of `mypy` wrongly inferring their return types to also include `int`.